### PR TITLE
Pd port

### PR DESCRIPTION
**Pull request by olbjo**
Draft implementation of reading and writing pd port records. 
Pushed for testing in "standard setup" with profinet switch
Diff accessways test should pass

### CODE REVIEW

**Comment by olbjo** [comment]
Updates included in this pr makes dcp alias and the different access ways tests pass.

The DAP  modules id definitions are moved from sample application ot pnet_api. This update has been rejected in previous prs.

I am not completely happy with these additions. I think port, interface and physical device concepts needs some additional thought.

@pyhys please rerun tests in real test setup and take a look at the updates

**Comment by olbjo** [comment]
Previous force-push fixes issue with different access ways test

File: src/device/pf_block_writer.c [hunk 1]
```diff
@@ -2400,3 +2413,464 @@ void pf_put_input_data(
...
+   }
+
+   /* Finally insert the block length into the block header */
+   block_len = *p_pos - (block_pos + 4);
+   block_pos += offsetof(pf_block_header_t, block_length); /* Point to correct place */
+   pf_put_uint16(is_big_endian, block_len, res_len, p_bytes, &block_pos);
+}
+
+void pf_put_pdport_statistics(
+    pnet_t                    *net,
```
Comment (pyhys) [inline src/device/pf_block_writer.c:2602]: Could we send in interface_statistics instead of net?

File: src/device/pf_block_writer.c [hunk 2]
```diff
@@ -2400,3 +2413,464 @@ void pf_put_input_data(
...
+    uint8_t                   *p_bytes,
+    uint16_t                  *p_pos)
+{
+   uint16_t block_pos = *p_pos;
+   uint16_t block_len = 0;
+   uint16_t temp_u16 = 0;
+   uint8_t numPeers = net->lldp_peer_req.ttl ? 1 : 0;
+
+   /*Do not add anything if there are no peer request was requested*/
+   if (0 == numPeers)
```
Comment (pyhys) [inline src/device/pf_block_writer.c:2431]: if (numPeers == 0)

File: src/device/pf_block_writer.c [hunk 3]
```diff
@@ -2400,3 +2413,464 @@ void pf_put_input_data(
    block_pos += offsetof(pf_block_header_t, block_length);   /* Point to correct place */
    pf_put_uint16(is_big_endian, block_len, res_len, p_bytes, &block_pos);
 }
+
+void pf_put_pdport_data_check(
+    pnet_t                    *net,
```
Comment (pyhys) [inline src/device/pf_block_writer.c:2418]: Could we send in lldp_peer_req instead?

File: src/device/pf_cmrdr.c [hunk 4]
```diff
@@ -391,58 +391,39 @@ int pf_cmrdr_rm_read_ind(
...
-         /* ToDo: Implement properly when LLDP is done */
-         /*
-          * BlockHeader, Padding, Padding, SlotNumber, SubslotNumber,
-          * { [CheckPeers], [CheckLineDelay], [CheckMAUType a], [CheckLinkState],
-          *   [CheckSyncDifference], [CheckMAUTypeDifference], [CheckMAUTypeExtension] b }
-          * a The implementation of the MAUType check shall take the AutoNegotiation settings into account. A diagnosis with severity fault shall be issued if an AutoNegotiation error is detected after the next LinkDown / LinkUp do to the used AutoNegotiation settings.
-          * b Only possible if CheckMAUType is part of the list
-          */
-         ret = 0;
+         if ((PNET_SLOT_DAP_IDENT == p_read_request->slot_number) &&
```
Comment (pyhys) [inline src/device/pf_cmrdr.c:394]: Avoid Yoda conditions

Comment (olbjo) [reply src/device/pf_cmrdr.c]: Fixed

File: src/device/pf_block_writer.c (hunk 1 above)
Comment (olbjo) [reply src/device/pf_block_writer.c]: Fixed

File: src/common/pf_dcp.c [hunk 5]
```diff
@@ -253,6 +253,7 @@ static int pf_dcp_put_block(
  * @param opt                 In:   Option key.
  * @param sub                 In:   Sub-option key.
  * @param request_is_identify In:   Usage in response to Identify request (skips some blocks)
+ * @param append_alias_name   In:   Addend alias name
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/common/pf_dcp.c]: Fixed

File: src/device/pf_block_writer.c [hunk 6]
```diff
@@ -2400,3 +2413,464 @@ void pf_put_input_data(
...
+   /* Length OwnerPortID */
+   pf_put_byte(strlen(net->fspm_cfg.lldp_cfg.port_id), res_len, p_bytes, p_pos);
+
+   /* OwnerPortID */
+   pf_put_mem(net->fspm_cfg.lldp_cfg.port_id, strlen(net->fspm_cfg.lldp_cfg.port_id), res_len, p_bytes, p_pos);
+
+   /* Number of Peers */
+   pf_put_byte(numPeers, res_len, p_bytes, p_pos);
+   /*Input only the number if it's there */
+   if (0 < numPeers)
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_block_writer.c]: Fixed

**Comment by olbjo** [comment]
I just realized I failed to remove some temp magic numbers. I will remove these and push an update

File: src/device/pf_block_writer.c (hunk 3 above)
Comment (olbjo) [reply src/device/pf_block_writer.c]: fixed

File: src/device/pf_block_writer.c (hunk 2 above)
Comment (olbjo) [reply src/device/pf_block_writer.c]: Fixed

File: src/device/pf_block_writer.h [hunk 7]
```diff
@@ -645,6 +645,103 @@ void pf_put_diag_data(
...
+   pf_check_peer_t         *check_peer,
+   bool                    is_big_endian,
+   pf_iod_read_result_t    *p_res,
+   uint16_t                res_len,
+   uint8_t                 *p_bytes,
+   uint16_t                *p_pos);
+
+/**
+ * Insert pd port data adjust block into a buffer.
+ * @param net              In:    Peer to peer boundary ToDo - Add support for other adjust properties
```
Comment (pyhys) [inline src/device/pf_block_writer.h:666]: p_peer_to_peer_boundary

**Review by pyhys — APPROVED** [review]

File: src/device/pf_block_writer.h (hunk 7 above)
Comment (olbjo) [reply src/device/pf_block_writer.h]: Fixed

**Pull request merged**